### DISCUSSION TITLE
0.1.0

### DISCUSSION THREAD

**Discussion by mhmukadam** [facebookresearch/theseus#255] [Announcements]
## What's Changed
* Add SO3 support by @luisenp in https://github.com/facebookresearch/theseus/pull/46
* Taoshaf.add so3 class by @fantaosha in https://github.com/facebookresearch/theseus/pull/65
* Add SO3 rotate and unrotate by @fantaosha in https://github.com/facebookresearch/theseus/pull/57
* add so3._hat_matrix_check() by @fantaosha in https://github.com/facebookresearch/theseus/pull/59
* Encapsulated data loading functions in tactile pushing example into a new class by @luisenp in https://github.com/facebookresearch/theseus/pull/51
* Added a TactilePoseEstimator class to easily create TheseusLayer for tactile pushing by @luisenp in https://github.com/facebookresearch/theseus/pull/52
* Refactor tactile pushing model interface by @luisenp in https://github.com/facebookresearch/theseus/pull/55
* Minor fixes 02/01/2022 by @luisenp in https://github.com/facebookresearch/theseus/pull/66
* add adjoint, hat and vee for SE3 by @fantaosha in https://github.com/facebookresearch/theseus/pull/68
* Add SE3.exp_map() and SE3.log_map() by @fantaosha in https://github.com/facebookresearch/theseus/pull/71
* Add SE3.compose() by @fantaosha in https://github.com/facebookresearch/theseus/pull/72
* add SE3.transform_from and SE3.transform_to by @fantaosha in https://github.com/facebookresearch/theseus/pull/80
* Updated to CircleCI's next-gen images. by @luisenp in https://github.com/facebookresearch/theseus/pull/89
* Updated README with libsuitesparse installation instructions. by @luisenp in https://github.com/facebookresearch/theseus/pull/90
* Added kwarg to NonlinearOptimizer.optimizer() for tracking error history by @luisenp in https://github.com/facebookresearch/theseus/pull/82
* Merge infos results for truncated backward modes by @luisenp in https://github.com/facebookresearch/theseus/pull/83
* Fix Issue 88 by @maurimo in https://github.com/facebookresearch/theseus/pull/97
* Fixed bug in Variable.update() that was breaking torch graph... by @luisenp in https://github.com/facebookresearch/theseus/pull/96
* Forced Gauss-Newton step for last iterations of truncated backward. by @luisenp in https://github.com/facebookresearch/theseus/pull/81
* Add automatic differentiation on the Lie group tangent space by @fantaosha in https://github.com/facebookresearch/theseus/pull/74
* Add rand() to LieGroup by @fantaosha in https://github.com/facebookresearch/theseus/pull/95
* Fix SO2 rotate and unrotate jacobian by @fantaosha in https://github.com/facebookresearch/theseus/pull/58
* Add projection for sparse Jacobian matrices by @fantaosha in https://github.com/facebookresearch/theseus/pull/98
* Add LieGroup Support for AutoDiffFunction by @fantaosha in https://github.com/facebookresearch/theseus/pull/99
* Add SE2.transform from() and fix shape bugs in SE2.transform_to() by @fantaosha in https://github.com/facebookresearch/theseus/pull/103
* Switch SE3.transform_from and SE3.transform_to by @fantaosha in https://github.com/facebookresearch/theseus/pull/104
* Enabled back ellipsoidal damping in LM with linear solvers support checks by @luisenp in https://github.com/facebookresearch/theseus/pull/87
* Changed name of pytest mark for CUDA extensions. by @luisenp in https://github.com/facebookresearch/theseus/pull/102
* Fix backpropagation bugs in SO3 and SE3 log_map by @fantaosha in https://github.com/facebookresearch/theseus/pull/109
* Add analytical jacobians for `LieGroup.exp_map` by @fantaosha in https://github.com/facebookresearch/theseus/pull/110
* error and errorsquaredNorm optional data by @jeffin07 in https://github.com/facebookresearch/theseus/pull/105
* Add analytical derivatives for `LieGroup.log_map()` by @fantaosha in https://github.com/facebookresearch/theseus/pull/114
* Refactor `SO3.to_quaternion` to fix backward bugs and improve the accuracy around `pi` by @fantaosha in https://github.com/facebookresearch/theseus/pull/116
* Change RobotModel.forward_kinematics() interface by @luisenp in https://github.com/facebookresearch/theseus/pull/94
* Added error handling for missing `matplotlib` and `omegaconf` installation when using `theg` by @luisenp in https://github.com/facebookresearch/theseus/pull/119
* Add jacobians argument to exp_map and log_map functions by @joeaortiz in https://github.com/facebookresearch/theseus/pull/122
* Add pose graph optimization example by @fantaosha in https://github.com/facebookresearch/theseus/pull/118
* Added 'secret' option to keep constant step size when using truncated… by @luisenp in https://github.com/facebookresearch/theseus/pull/130
* Add jacobians for `LieGroup.local()` by @fantaosha in https://github.com/facebookresearch/theseus/pull/129
* Added method to update SE2 from x_y_theta. by @luisenp in https://github.com/facebookresearch/theseus/pull/131
* A complete version of Bundle Adjusment  by @luisenp in https://github.com/facebookresearch/theseus/pull/117
* Fixed jacobians for Between and VariableDifference by @fantaosha in https://github.com/facebookresearch/theseus/pull/133
* Added batching support to tactile pushing example. by @luisenp in https://github.com/facebookresearch/theseus/pull/132
* Changed tactile pose example optim var initialization to use start pose for all vars by @luisenp in https://github.com/facebookresearch/theseus/pull/137
* Override Vector operators in Point2 and Point3 by @jeffin07 in https://github.com/facebookresearch/theseus/pull/124
* Merge `Between` and `VariableDiff` with `RelativePoseError` and `PosePriorError` by @fantaosha in https://github.com/facebookresearch/theseus/pull/136
* Fix a bug in Objective.copy() by @fantaosha in https://github.com/facebookresearch/theseus/pull/139
* Added option to force max iterations for TactilePoseEstimator. by @luisenp in https://github.com/facebookresearch/theseus/pull/141
* black version bump by @jeffin07 in https://github.com/facebookresearch/theseus/pull/144
* Added code to split tactile pushing trajectories data into train/val by @luisenp in https://github.com/facebookresearch/theseus/pull/143
* Removed RobotModel.dim() by @luisenp in https://github.com/facebookresearch/theseus/pull/156
* [bug-fix] Fixed wrong data shape initialization for GPCostWeight.dt by @luisenp in https://github.com/facebookresearch/theseus/pull/157
* Made AutoDiffCostFunction._tmp_optim_vars copies of original by @luisenp in https://github.com/facebookresearch/theseus/pull/155
* Add forward kinematics using an URDF to theseus.embodied.kinematics. by @exhaustin in https://github.com/facebookresearch/theseus/pull/84
* Fixed dtype error in `se3.py` that came up in unit tests by @joeaortiz in https://github.com/facebookresearch/theseus/pull/158
* Add-ons for backward experiments on Tactile Pose Estimation by @luisenp in https://github.com/facebookresearch/theseus/pull/164
* Change unit tests to avoid making mypy a main requirement. by @luisenp in https://github.com/facebookresearch/theseus/pull/168
* Update readme and contrib by @luisenp in https://github.com/facebookresearch/theseus/pull/169
* Add ManifoldGaussian class for messages in belief propagation by @joeaortiz in https://github.com/facebookresearch/theseus/pull/121
* More efficient implementation of forward kinematics by @exhaustin in https://github.com/facebookresearch/theseus/pull/175
* Changing setup virtualenv command. by @luisenp in https://github.com/facebookresearch/theseus/pull/178
* Updated SDF object in collision cost functions whenever an aux var is updated by @luisenp in https://github.com/facebookresearch/theseus/pull/177
* Fixed device bug that occurred when merging info in TRUNCATED backward modes by @luisenp in https://github.com/facebookresearch/theseus/pull/181
* Allow constant inputs to cost functions to be passed as floats by @jeffin07 in https://github.com/facebookresearch/theseus/pull/150
* Minor changes to core test code. by @luisenp in https://github.com/facebookresearch/theseus/pull/197
* adding logo by @mhmukadam in https://github.com/facebookresearch/theseus/pull/200
* Added aliases for Difference and Between by @luisenp in https://github.com/facebookresearch/theseus/pull/199
* Fixed infinite recursion in GPMotionModel.copy() by @luisenp in https://github.com/facebookresearch/theseus/pull/201
* Fix bug in diagonal cost weight by @luisenp in https://github.com/facebookresearch/theseus/pull/203
* Added a check in TheseusFunction that enforces copy() also copies the variables by @luisenp in https://github.com/facebookresearch/theseus/pull/202
* Fixed bug in Objective.error() that was updating data unncessarily. by @luisenp in https://github.com/facebookresearch/theseus/pull/204
* DLM gradients by @rtqichen in https://github.com/facebookresearch/theseus/pull/161
* Setup now uses torch for checking CUDA availability, and CI runs py3.9 tests by @luisenp in https://github.com/facebookresearch/theseus/pull/206
* Update README to specify python versions and CUDA during install step + by @cpaxton in https://github.com/facebookresearch/theseus/pull/207
* Vectorization refactor by @luisenp in https://github.com/facebookresearch/theseus/pull/205
* Implement Robust Cost Function by @luisenp in https://github.com/facebookresearch/theseus/pull/148
* Added option for auto resetting LUCudaSparseSolver if the batch size needs to change by @luisenp in https://github.com/facebookresearch/theseus/pull/212
* Address comments on RobustCostFunction implementation by @luisenp in https://github.com/facebookresearch/theseus/pull/213
* Moved the method that retracts all variables with a given delta to Objective by @luisenp in https://github.com/facebookresearch/theseus/pull/214
* Fixed flaky unit test for Collision2D jacobians. by @luisenp in https://github.com/facebookresearch/theseus/pull/216
* Bundle Adjustment using RobustCostFunction by @luisenp in https://github.com/facebookresearch/theseus/pull/149
* Fixed the cross product bug for SE3.exp_map and SE3.log_map by @fantaosha in https://github.com/facebookresearch/theseus/pull/217
* Vectorize optimization variables retraction step  by @luisenp in https://github.com/facebookresearch/theseus/pull/215
* Moved Vectorize(objective) to the Optimizer class. by @luisenp in https://github.com/facebookresearch/theseus/pull/218
* Added on-demand vectorization and also vectorized Objective.error(). by @luisenp in https://github.com/facebookresearch/theseus/pull/221
* Vectorize PGO by @luisenp in https://github.com/facebookresearch/theseus/pull/211
* Destroy cusolver context in CusolverLUSolver destructor by @luisenp in https://github.com/facebookresearch/theseus/pull/222
* Jacobian computation using loop  by @fantaosha in https://github.com/facebookresearch/theseus/pull/225
* Vectorization handles singleton costs by @luisenp in https://github.com/facebookresearch/theseus/pull/226
* Added close-formed jacobian for DLM perturbation cost. by @luisenp in https://github.com/facebookresearch/theseus/pull/224
* SE2/SE3/SO3 - consolidate EPS, add dtype-conditioned EPS and add float32 unit tests by @luisenp in https://github.com/facebookresearch/theseus/pull/220
* Add normalization to Lie group by @fantaosha in https://github.com/facebookresearch/theseus/pull/227
* Add normalization to Lie group constructor by @fantaosha in https://github.com/facebookresearch/theseus/pull/228
* Benchmarking PGO on main branch by @fantaosha in https://github.com/facebookresearch/theseus/pull/233
* Renamed `Variable.data` as `Variable.tensor` by @luisenp in https://github.com/facebookresearch/theseus/pull/229
* More data -> tensor renaming by @luisenp in https://github.com/facebookresearch/theseus/pull/230
* Added state history tracking by @luisenp in https://github.com/facebookresearch/theseus/pull/234
* Unified all cost functions so that cost weight is the last non-default argument by @luisenp in https://github.com/facebookresearch/theseus/pull/235
* Ensure that CHOLMOD python interface casts to 64-bit precision by @luisenp in https://github.com/facebookresearch/theseus/pull/238
* Add sphinx and readthedocs configuration by @luisenp in https://github.com/facebookresearch/theseus/pull/237
* Fixed bug in state history for matrix data tensors. by @luisenp in https://github.com/facebookresearch/theseus/pull/240
* Added isort for examples folder by @luisenp in https://github.com/facebookresearch/theseus/pull/243
* Avoid batching SDF data, since it's shared by all trajectories. by @luisenp in https://github.com/facebookresearch/theseus/pull/246
* Fixed device bug in DLM perturbation's jacobians by @luisenp in https://github.com/facebookresearch/theseus/pull/247
* Benchmark PGO on the main branch by @fantaosha in https://github.com/facebookresearch/theseus/pull/244
* Added checks to enforce 32- or 64-bit dtype by @luisenp in https://github.com/facebookresearch/theseus/pull/245
* Update readme by @mhmukadam in https://github.com/facebookresearch/theseus/pull/251
* Added MANIFEST.in and changed project name to theseus-ai. by @luisenp in https://github.com/facebookresearch/theseus/pull/252
* Adding simple example by @mhmukadam in https://github.com/facebookresearch/theseus/pull/253
* Added option to clear cuda cache when vectorization cache is cleared. by @luisenp in https://github.com/facebookresearch/theseus/pull/249
* Added evaluation directory by @luisenp in https://github.com/facebookresearch/theseus/pull/241

## New Contributors
* @jeffin07 made their first contribution in https://github.com/facebookresearch/theseus/pull/105
* @joeaortiz made their first contribution in https://github.com/facebookresearch/theseus/pull/122
* @exhaustin made their first contribution in https://github.com/facebookresearch/theseus/pull/84
* @rtqichen made their first contribution in https://github.com/facebookresearch/theseus/pull/161
* @cpaxton made their first contribution in https://github.com/facebookresearch/theseus/pull/207

**Full Changelog**: https://github.com/facebookresearch/theseus/compare/0.1.0-b.2...0.1.0

<hr /><em>This discussion was created from the release <a href='https://github.com/facebookresearch/theseus/releases/tag/0.1.0'>0.1.0</a>.</em>
